### ISSUE TITLE
Editor doesn't render on android device with ionic.

### ISSUE THREAD

**Issue by yahmez00** [KillerCodeMonkey/ngx-quill#560]
Hey there.

First off, love your work with the plugin, it works great apart from this issue I'm having.

The editor appears and works just fine in my browser, and even on my mobile browser (Chrome, Firefox, Safari), however when I deploy my ionic app either to my actual android device, or to an emulated device, the editor fails to render/appear. An invisible text field still appears that you can type into, but it's obviously not working as it should.

I thought it might be to do with my custom CSS, but I commented all that and reverted it to the default and I had the same problem. I've tried to emulate your ionic setup as much as possible but I'm at a loss as to why it's not working. Seems like some file(s) may not be loading correctly, but there are no debug messages to work off unfortunately.

Here's what it should look like:
<img width="299" alt="quill_correct" src="https://user-images.githubusercontent.com/5257788/66211952-572b9100-e700-11e9-97b0-a22690df1de5.png">

And here's how it looks on a real/emulated device:
![quill_broken](https://user-images.githubusercontent.com/5257788/66211950-5692fa80-e700-11e9-94d7-b5bc8aa52c60.jpg)

Any help would be appreciated, thanks!

**Comment by KillerCodeMonkey** [comment]
Puh never tried IT on a real device. Maybe i can Run my ionic example Repo
in my Android device and See If IT IS working .

In General quilljs ist Not optimized for native mobile Apps. So maybe If it
renders correctly there can be Others usability issues.

So You use Ionic 4 + angular 8?

James Officer <notifications@github.com> schrieb am Fr., 4. Okt. 2019,
15:46:

**Comment by KillerCodeMonkey** [comment]
tried it on emulator and on my android device:

<img width="361" alt="Bildschirmfoto 2019-10-04 um 20 20 48" src="https://user-images.githubusercontent.com/2264672/66231699-6520e800-e6e7-11e9-9220-d04793623b89.png">

<img width="361" alt="photo_2019-10-04_20-40-19" src="https://user-images.githubusercontent.com/2264672/66231717-6fdb7d00-e6e7-11e9-81f7-861c7c26d476.jpg">

Works like charme, so i think it has something to do with your build process. Maybe some modules are not packed in your final bundle or you have submodules and missed the import of the QuillModule there.

it is working with production build `ionic cordova run android --device --prod` and dev build `ionic cordova run android --device`

**Comment by yahmez00** [comment]
I currently have my quill setup in a `shared.module.ts` module. That module shares components across the app. Everything else in there seems to shares fine (such as Google Maps), seems to only be quill that doesn't want to work which is very odd. Everything also works fine on a normal `--prod` build (to my web server), only when I emulate to Android that it doesn't work.

My module looks like this currently (trimmed out some other stuff for brevity):

```typescript
@NgModule({
    imports: [
        CommonModule,
        IonicModule,
        RouterModule,
        FormsModule,
        ReactiveFormsModule,
        QuillModule.forRoot({
            modules: {
                syntax: false,
                toolbar: [
                    ['bold', 'italic', 'underline', 'strike'],
                    [{ list: 'ordered' }, { list: 'bullet' }],
                ],
            },
        }),
    ],
    declarations: [],
    exports: [
        ...
        QuillModule,
    ],
})
export class SharedModule {}
```

I'm struggling to think what else it could be. I'm still using Angular 7, but doubt that would make much of a difference, I also updated to the latest `ngx-quill` as I was only on v4 before.

Will let you know if I discover anything.

**Comment by KillerCodeMonkey** [comment]
in general you would do the "QuillModule.forRoot()" stuff in your main module (app.module).

in all other modules you should only import "QuillModule".

Only then you can share a "global" config - that is what .forRoot does.

Try it this way

**Comment by KillerCodeMonkey** [comment]
did you solved it?

**Comment by yahmez00** [comment]
Figured it out! The problem was that I was loading in the `quill.snow.css` stylesheet through the cdn in my `index.html` file. I removed it out of there, put it into it's own stylesheet and then put that into my `angular.js` file and now everything is working. I'm assuming this was because when running on the emulator or my device it wasn't able to connect to the cdn which meant the styles broke.

I also followed your tip on the module too, I moved the `forRoot` stuff into `app.module` as well. Cheers for that. And thanks for your help and patience!
